src/components/navigation/www/navigation.jsx

```diff
@@ -27,35 +27,27 @@ class Navigation extends React.Component {
         super(props);
         bindAll(this, [
             'getProfileUrl',
-            'handleSearchSubmit'
+            'handleSearchSubmit',
+            'setupMessagePolling'
         ]);
         this.state = {
             messageCountIntervalId: -1 // javascript method interval id for getting messsage count.
```

Comment:
this is maybe going beyond the scope of this change, but I don't think that messageCountIntervalId belongs inside state. There's no need for the component to re-render just because messageCountIntervalId changed. 

There are a few examples of how you might just store the response from `setTimeout()` in a regular member variable: in www, src/components/modal/social/container.jsx does this, and in gui, src/components/action-menu/action-menu.jsx does this.

But I totally understand if you don't want to make that change as part of this change!

Reply:
I was swamped this week so didn't have time to look into this. If you don't have time to review before you head out of office, maybe I'll take a look next week. Otherwise, I suggest wee keep it for a follow up.